assets/javascripts/discourse/components/d-templates/modal/form.js

```diff
@@ -1,9 +1,11 @@
 import Component from "@glimmer/component";
-import { getOwner } from "discourse-common/lib/get-owner";
 
 export default class DTemplates extends Component {
-  dataModel =
-    this.args.model?.model ||
-    getOwner(this).lookup("controller:composer").model;
-  textarea = this.args.model?.textarea;
+  get onInsertTemplate() {
+    return this.args.model?.onInsertTemplate;
+  }
+
+  get textarea() {
+    return this.args.model?.textarea;
+  }
```

Comment:
Both args can be accessed directly in the template (`{{@model.onInsertTemplate}}`, `{{@model.textarea}}`) and then this file could be removed.

Reply:
Done